analyzers/src/SonarAnalyzer.CSharp/Rules/MultilineBlocksWithoutBrace.cs

```diff
@@ -28,143 +28,118 @@
 
-        private static readonly DiagnosticDescriptor rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat);
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c => CheckLoop(c, ((WhileStatementSyntax)c.Node).Statement),
-                SyntaxKind.WhileStatement);
+    protected override void Initialize(SonarAnalysisContext context)
+    {
+        context.RegisterSyntaxNodeActionInNonGenerated(
+            c => CheckLoop(c, ((WhileStatementSyntax)c.Node).Statement),
+            SyntaxKind.WhileStatement);
 
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c => CheckLoop(c, ((ForStatementSyntax)c.Node).Statement),
-                SyntaxKind.ForStatement);
+        context.RegisterSyntaxNodeActionInNonGenerated(
+            c => CheckLoop(c, ((ForStatementSyntax)c.Node).Statement),
+            SyntaxKind.ForStatement);
 
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c => CheckLoop(c, ((ForEachStatementSyntax)c.Node).Statement),
-                SyntaxKind.ForEachStatement);
+        context.RegisterSyntaxNodeActionInNonGenerated(
+            c => CheckLoop(c, ((ForEachStatementSyntax)c.Node).Statement),
+            SyntaxKind.ForEachStatement);
 
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c => CheckIf(c, (IfStatementSyntax)c.Node),
-                SyntaxKind.IfStatement);
-        }
+        context.RegisterSyntaxNodeActionInNonGenerated(
+            c => CheckIf(c, (IfStatementSyntax)c.Node),
+            SyntaxKind.IfStatement);
+    }
 
-        private static void CheckLoop(SyntaxNodeAnalysisContext context, StatementSyntax statement)
+    private static void CheckLoop(SyntaxNodeAnalysisContext context, StatementSyntax statement)
+    {
+        if (!IsNestedStatement(statement))
         {
-            if (IsNestedStatement(statement))
-            {
-                return;
-            }
-
             CheckStatement(context, statement, "in a loop", "only once");
         }
+    }
 
-        private static void CheckIf(SyntaxNodeAnalysisContext context, IfStatementSyntax ifStatement)
+    private static void CheckIf(SyntaxNodeAnalysisContext context, IfStatementSyntax ifStatement)
+    {
+        if (!ifStatement.GetPrecedingIfsInConditionChain().Any()
+            && !IsNestedStatement(ifStatement.Statement)
+            && GetLastStatementInIfChain(ifStatement) is { } lastStatementInIfChain
```

Comment:
Can this return `null`? I suspect it can and will cause an issue with coverage. Please try to add a test case, something like this could do the trick:
```
    void WithError(bool condition)
    {
        if (condition)
            // Error@-1 [CS1002] ; expected
            // Error@-2 [CS1525] Invalid expression term
    }
```